File: src/main/java/to7fa/entities/evenement.java
package to7fa.entities;

import java.sql.Date;

/**
 *
 * @author Admin
 */
public class evenement {

    private int ID_event;
    private String nom_event;
    private String description_event;
    private String lieu_event;
    private String type_event;
    private Date dateDebut_event;
    private Date dateFin_event;
    private int capacite_event;
    private String image_event;
    private double prix_event;

    public evenement() {
    }

    public evenement(int ID_event, String nom_event, String description_event, String lieu_event, String type_event, Date dateDebut_event, Date dateFin_event, int capacite_event,String image_event, double prix_event) {
        this.ID_event = ID_event;
        this.nom_event = nom_event;
        this.description_event = description_event;
        this.lieu_event = lieu_event;
        this.type_event = type_event;
        this.dateDebut_event = dateDebut_event;
        this.dateFin_event = dateFin_event;
        this.capacite_event = capacite_event;
        this.image_event = image_event;
        this.prix_event = prix_event;
    }

    public evenement(String nom_event, String description_event, String lieu_event, String type_event, Date dateDebut_event, Date dateFin_event, int capacite_event, String image_event, double prix_event) {
        this.nom_event = nom_event;
        this.description_event = description_event;
        this.lieu_event = lieu_event;
        this.type_event = type_event;
        this.dateDebut_event = dateDebut_event;
        this.dateFin_event = dateFin_event;
        this.capacite_event = capacite_event;
        this.image_event = image_event;
        this.prix_event = prix_event;
    }

    public int getID_event() {
        return ID_event;
    }

    public void setID_event(int ID_event) {
        this.ID_event = ID_event;
    }

    public String getNom_event() {
        return nom_event;
    }

    public void setNom_event(String nom_event) {
        this.nom_event = nom_event;
    }

    public String getDescription_event() {
        return description_event;
    }

    public void setDescription_event(String description_event) {
        this.description_event = description_event;
    }

    public String getLieu_event() {
        return lieu_event;
    }

    public void setLieu_event(String lieu_event) {
        this.lieu_event = lieu_event;
    }

    public String getType_event() {
        return type_event;
    }

    public void setType_event(String type_event) {
        this.type_event = type_event;
    }

    public Date getDateDebut_event() {
        return dateDebut_event;
    }

    public void setDateDebut_event(Date dateDebut_event) {
        this.dateDebut_event = dateDebut_event;
    }

    public Date getDateFin_event() {
        return dateFin_event;
    }

    public void setDateFin_event(Date dateFin_event) {
        this.dateFin_event = dateFin_event;
    }

    public int getCapacite_event() {
        return capacite_event;
    }

    public void setCapacite_event(int capacite_event) {
        this.capacite_event = capacite_event;
    }


    public String getImage_event() {
        return image_event;
    }

    public void setImage_event(String image_event) {
        this.image_event = image_event;
    }

    public double getPrix_event() {
        return prix_event;
    }

    public void setPrix_event(double prix_event) {
        this.prix_event = prix_event;
    }

    @Override
    public String toString() {
        return "evenement{" + "ID_event=" + ID_event + ", nom_event=" + nom_event + ", description_event=" + description_event + ", lieu_event=" + lieu_event + ", type_event=" + type_event + ", dateDebut_event=" + dateDebut_event + ", dateFin_event=" + dateFin_event + ", capacite_event=" + capacite_event + ", image_event=" + image_event + ", prix_event=" + prix_event + '}';
    }



}

File: src/Entity/Evenement.php
<?php

namespace App\Entity;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;


/**
 * Evenement
 *
 * @ORM\Table(name="evenement")
 * @ORM\Entity
 */
class Evenement
{
    /**
     * @var int
     *
     * @ORM\Column(name="id_event", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idEvent;

    /**
     * @var string
     *
     * @ORM\Column(name="nom_event", type="string", length=255, nullable=false)
     * @Assert\NotBlank(message="The event name cannot be empty.")
     * @Assert\Length(max=255)
     */
    private $nomEvent;

    /**
     * @var string
     *
     * @ORM\Column(name="description_event", type="string", length=255, nullable=false)
     * @Assert\NotBlank
     * @Assert\Length(min=10, max=255)
     */
    private $descriptionEvent;

    /**
     * @var string
     *
     * @ORM\Column(name="lieu_event", type="string", length=255, nullable=false)
     * @Assert\NotBlank
     * @Assert\Length(max=255)
     */
    private $lieuEvent;

    /**
     * @var string
     *
     * @ORM\Column(name="type_event", type="string", length=50, nullable=false)
     * @Assert\NotBlank
     * @Assert\Length(max=255)
     */
    private $typeEvent;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="dateDebut_event", type="date", nullable=false)
     * @Assert\NotNull
     */
    private $datedebutEvent;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="dateFin_event", type="date", nullable=false)
     * @Assert\NotNull
     * @Assert\GreaterThan(propertyPath="datedebutEvent")
     */
    private $datefinEvent;

    /**
     * @var int
     *
     * @ORM\Column(name="capacite_event", type="integer", nullable=false)
     * @Assert\NotNull
     * @Assert\Positive
     */
    private $capaciteEvent;

    /**
     * @var string
     *
     * @ORM\Column(name="image_event", type="string", length=255, nullable=false)
     */
    private $imageEvent;

    /**
     * @var float
     *
     * @ORM\Column(name="prix_event", type="float", precision=10, scale=0, nullable=false)
     * @Assert\NotBlank
     * @Assert\PositiveOrZero
     */
    private $prixEvent;

    /**
     * @ORM\OneToMany(targetEntity=Participation::class, mappedBy="evenement", orphanRemoval=true)
     */
    private $participations;

    public function __construct()
    {
        $this->participations = new ArrayCollection();
    }

    public function getIdEvent(): ?int
    {
        return $this->idEvent;
    }

    public function getNomEvent(): ?string
    {
        return $this->nomEvent;
    }

    public function setNomEvent(string $nomEvent): self
    {
        $this->nomEvent = $nomEvent;

        return $this;
    }

    public function getDescriptionEvent(): ?string
    {
        return $this->descriptionEvent;
    }

    public function setDescriptionEvent(string $descriptionEvent): self
    {
        $this->descriptionEvent = $descriptionEvent;

        return $this;
    }

    public function getLieuEvent(): ?string
    {
        return $this->lieuEvent;
    }

    public function setLieuEvent(string $lieuEvent): self
    {
        $this->lieuEvent = $lieuEvent;

        return $this;
    }

    public function getTypeEvent(): ?string
    {
        return $this->typeEvent;
    }

    public function setTypeEvent(string $typeEvent): self
    {
        $this->typeEvent = $typeEvent;

        return $this;
    }

    public function getDatedebutEvent(): ?\DateTimeInterface
    {
        return $this->datedebutEvent;
    }

    public function setDatedebutEvent(\DateTimeInterface $datedebutEvent): self
    {
        $this->datedebutEvent = $datedebutEvent;

        return $this;
    }

    public function getDatefinEvent(): ?\DateTimeInterface
    {
        return $this->datefinEvent;
    }

    public function setDatefinEvent(\DateTimeInterface $datefinEvent): self
    {
        $this->datefinEvent = $datefinEvent;

        return $this;
    }

    public function getCapaciteEvent(): ?int
    {
        return $this->capaciteEvent;
    }

    public function setCapaciteEvent(int $capaciteEvent): self
    {
        $this->capaciteEvent = $capaciteEvent;

        return $this;
    }

    public function getImageEvent(): ?string
    {
        return $this->imageEvent;
    }

    public function setImageEvent(string $imageEvent): self
    {
        $this->imageEvent = $imageEvent;

        return $this;
    }

    public function getPrixEvent(): ?float
    {
        return $this->prixEvent;
    }

    public function setPrixEvent(float $prixEvent): self
    {
        $this->prixEvent = $prixEvent;

        return $this;
    }

    /**
     * @return Collection<int, Participation>
     */
    public function getParticipations(): Collection
    {
        return $this->participations;
    }

    public function addParticipation(Participation $participation): self
    {
        if (!$this->participations->contains($participation)) {
            $this->participations[] = $participation;
            $participation->setEvenement($this);
        }

        return $this;
    }

    public function removeParticipation(Participation $participation): self
    {
        if ($this->participations->removeElement($participation)) {
            // set the owning side to null (unless already changed)
            if ($participation->getEvenement() === $this) {
                $participation->setEvenement(null);
            }
        }

        return $this;
    }

    public function __toString()
    {
        return $this->nomEvent;
    }
}
